Sources/CacheAdvance/CacheAdvanceErrors.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  Created by Dan Federman on 11/10/19.
+//  Copyright © 2019 Dan Federman.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+public enum CacheAdvanceReadError: Error, Equatable {
+    /// Thrown when the cache file is of an unexpected format.
+    /// A corrupted file should be deleted. Corruption can occur if an application crashes while writing to the file.
```

Comment:
> Corruption can occur if an application crashes while writing to the file.

From a comment in `CacheAdvance.swift` it seems like we are writing atomically. Is this true still?

Reply:
We are writing atomically. But it's still possible for an application to crash while we're writing (trivial example: while we're writing, the app divides by zero on another thread)